x/auth/ante/basic.go

```diff
@@ -18,15 +19,20 @@ import (
 // If ValidateBasic passes, decorator calls next AnteHandler in chain. Note,
 // ValidateBasicDecorator decorator will not get executed on ReCheckTx since it
 // is not dependent on application state.
-type ValidateBasicDecorator struct{}
+type ValidateBasicDecorator struct {
+	ak AccountKeeper
```

Comment:
why not pass in environment instead?

Reply:
yeah works, let me do that instead

Reply:
I had done it for consistency with other ante handlers as they take auth.